spec/syntax.md

```diff
@@ -526,18 +528,73 @@ unrecognized reserved sequences have no meaning and MAY result in errors during
 
 ```abnf
 reserved       = reserved-start reserved-body
-reserved-start = "!" / "@" / "#" / "%" / "*" / "<" / ">" / "/" / "?" / "~"
+reserved-start = "!" / "#" / "%" / "*" / "<" / ">" / "/" / "?" / "~"
 
 reserved-body  = *( [s] 1*(reserved-char / reserved-escape / quoted))
 reserved-char  = %x00-08        ; omit HTAB and LF
                / %x0B-0C        ; omit CR
                / %x0E-19        ; omit SP
-               / %x21-5B        ; omit \
+               / %x21-3F        ; omit @
+               / %x41-5B        ; omit \
                / %x5D-7A        ; omit { | }
                / %x7E-D7FF      ; omit surrogates
                / %xE000-10FFFF
 ```
 
+### Expression Attributes
+
+An **_<dfn>expression attribute</dfn>_** is a key-value pair
+containing a named argument that is attached to the _expression_ as a whole.
+They are intended to convey information about the _expression_
+that is not necessarily applicable as a _function_ _option_.
+For example, the `@locale` attribute overrides the locale of the _message_ for its _expression_.
+Attributes like `@can-delete` and `@translate` might not have any impact on formatting,
+acting instead as instructions to translators.
+
+_Expression attributes_ are prefixed by a U+0040 COMMERCIAL AT `@` sign,
+but otherwise use the same syntax as _options_.
+
+The _name_ of an _expression attribute_ MUST either be one of the following,
+or it MUST contain at least one U+002D HYPHEN-MINUS `-` character:
```

Comment:
I think you're trying to create a reserved namespace of "no hyphens" for use by the specification and then allow implementations to define their own expression attributes by forcing them to include hyphens. If that's the case, _we should say that_. Personally, I would have gone for some other mechanism, such as reserving `_` (which is a valid `name` starter) for implementation defined items, e.g.:

> {$foo :function option=value @locale=fr-FR @canReorder=no @_custom=val}

Reply:
The reason for creating the namespace is explicitly said a few lines further down:
> To ensure future extensibility of this specification,
all other _expression attributes_ with a _name_ not containing `-` MUST be ignored.

An underscore `_` is a possible alternative as well. I went with `-` because that's the [requirement](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name) used by custom HTML elements, and that seemed like pretty decent prior art to follow.

Reply:
The starts-with-underscore I find unattractive, but it fits with `name` without any changes and only requires a one-character lookahead to figure it out. I like `-` okay, I guess, but we really should resolve our naming issues so that we don't have a motley collection of naming schemes.

I filed #453 to track.